File: Busca e ordenação/Insertion Sort/insertion_sort.py
def inertion_sort(lista):
    tamanho = len(lista)
    for i in range(1, tamanho):
        chave = lista[i]
        j = i - 1
        while j >= 0 and lista[j] > chave:
            lista[j+1] = lista[j]
            j = j - 1
        lista[j+1] = chave

vector = [7, 5, 1, 8, 3, 9, 15, 2]
print("Vetor antes da ordenação: ", vector)

inertion_sort(vector)
print("Vetor depois da ordenação: ", vector)

File: Busca e ordenação/Insertion Sort/InsertionSort/src/insertionsort/InsertionSort.java
package insertionsort;

public class InsertionSort {
    static void insertion_sort(int vet[]) {
        int i, j, chave, tamanho = vet.length;
        
        for(i = 1; i < tamanho; i++) {
            chave = vet[i];
            j = i-1;
            while(j >= 0 && vet[j] > chave) {
                vet[j+1] = vet[j];
                j -= 1;
            }
            vet[j+1] = chave;
        }
    }
    
    static void print_vetor(int[] vet) {
        for(int elemento : vet) {
            System.out.printf("%d ", elemento);
        }
        System.out.print("\n");
    }
    
    public static void main(String[] args) {
        int vetor[] = {7, 5, 1, 8, 3, 9, 15, 2};
        
        System.out.print("Vetor antes da ordenação: ");
        print_vetor(vetor);
        
        insertion_sort(vetor);
        System.out.print("Vetor depois da ordenação:");
        print_vetor(vetor);
    }
    
}
